libs/output-mapping/src/Configuration/Table/Webalizer.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\OutputMapping\Configuration\Table;
+
+use Keboola\OutputMapping\Writer\Helper\RestrictedColumnsHelper;
+use Keboola\StorageApi\Client;
+
+class Webalizer
+{
+    public function __construct(readonly private Client $client)
+    {
+    }
+
+    public function webalize(array $configuration): array
+    {
+        if (isset($configuration['columns'])) {
+            $configuration['columns'] = $this->getWebalizedColumnNames($configuration['columns']);
+        }
+
+        if (isset($configuration['primaryKey'])) {
+            $configuration['primaryKey'] = $this->getWebalizedColumnNames($configuration['primaryKey']);
+        }
+
+        if (isset($configuration['column_metadata'])) {
+            $configuration['column_metadata'] = $this->webalizeColumnsMetadata($configuration['column_metadata']);
+        }
+
+        if (isset($configuration['schema'])) {
+            $configuration['schema'] = $this->webalizeSchemaColumns($configuration['schema']);
+        }
+
+        return $configuration;
+    }
+
+    private function webalizeColumnsMetadata(array $columnsMetadata): array
+    {
+        $columns = array_keys($columnsMetadata);
+        $columns = array_combine($columns, $this->getWebalizedColumnNames($columns));
+        foreach ($columnsMetadata as $columnName => $metadata) {
+            unset($columnsMetadata[$columnName]);
+            $columnsMetadata[$columns[$columnName]] = $metadata;
+        }
+
+        return $columnsMetadata;
+    }
+
+    private function webalizeSchemaColumns(array $schema): array
+    {
+        $schemaNames = array_map(fn($v) => $v['name'], $schema);
+
+        $schemaNames = array_combine($schemaNames, $this->getWebalizedColumnNames($schemaNames));
+
+        foreach ($schema as $k => $item) {
+            $schema[$k]['name'] = $schemaNames[$item['name']];
+        }
+        return $schema;
+    }
+
+    private function getWebalizedColumnNames(array $columns): array
+    {
+        $webalized = $this->client->webalizeColumnNames($columns);
+        $webalized = $webalized['columnNames'];
+
+        foreach ($columns as $k => $column) {
+            if (!RestrictedColumnsHelper::isRestrictedColumn($column)) {
+                $webalized[$k] = $column;
+            }
```

Comment:
Hm, jako na 3 pokus jsem to pobral jak to funguje. Jako je to asi dobrý, ale dost tricky - tím, že se restricted column nezwebalizuje, tak zůstane jak je a pak dál ho chytí restrcited columns helper/validator whatever a něco s tím udělá. Aspoň by tu teda měl být komentář v tom smyslu, proč to tady nevyhazuje výjimku (což bych čekal na první dobrou)

Reply:
no tak jasně, taky by to tak šlo udělat že když to je restricted column tak to vyhodí vyjímku...ale

neudělal jsem to tak z toho důvodu že to není starost webalizace ale validace... nechtěl jsem aby webalizace něco validovala
na druhou stranu že restricted columns nechám tak jak jsou je celkem legitimní - prostě systémové sloupečky se webalizovat nesmí

Reply:
Jo, pochopil jsem to tak, proto pisu, ze by to chtelo jen koment ve smyslu, proc tam ta exception neni (aby ju tam treba nekdo za 1/2 roku nepridal)